cpp/tests/io/json_tree.cpp

```diff
@@ -235,10 +235,8 @@ tree_meta_t2 get_tree_representation_cpu(
 {
   constexpr bool include_quote_char = true;
   // Copy the JSON tokens to the host
-  thrust::host_vector<cuio_json::PdaTokenT> tokens =
-    cudf::detail::make_host_vector_async(tokens_gpu, stream);
-  thrust::host_vector<cuio_json::SymbolOffsetT> token_indices =
-    cudf::detail::make_host_vector_async(token_indices_gpu1, stream);
+  auto tokens        = cudf::detail::make_host_vector_async(tokens_gpu, stream);
```

Comment:
This is a fun one. Here what I think happened here: this broke when I changed the return type because it would copy the buffer as a part of the implicit conversion(`cudf::detail::host_vector` -> `thrust::host_vector`); however, the data in the original object would not be ready because of the async D2H copy.